### DISCUSSION TITLE
"Complex" Katex formulas can't be rendered using MDX

### DISCUSSION THREAD

**Discussion by nikoladev** [mdx-js/mdx#1329] [Q&A] [answered]
I don't know the exact terms to describe it, but "complex" Katex formulas prevent MDX from rendering. (Feel free to change the title into something more specific)

I've figured out a workaround by passing the formula as a string to a custom component that runs the string through `katex.renderToString` and then passes the output to `dangerouslySetInnerHTML`. That might help in pinpointing where the problem occurs.

There is a [minimal reproducible example here](https://github.com/nikoladev/mdx-katex-bug) with instructions on how to make it fail as well as the workaround in action.

**Comment by ChristianMurphy** [comment, marked as answer]
Sorry you run into some trouble @nikoladev 

Looking to further reduce the problem I tried
Katex + MDX 1 runtime: https://codesandbox.io/s/katex-example-mdx-1-937f6
and Katex + MDX 2 runtime: https://codesandbox.io/s/katex-example-mdx-2-em1jg
Both worked :thinking: 

`gatsby-plugin-mdx` has forked parts of the MDX pipeline, which has been problematic in the past.
It's managed by the Gatsby team independent of the MDX team https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-plugin-mdx
I suspect, this has something to do with `gatsby-plugin-mdx`

**Reply by nikoladev** [reply]
Thanks for the help again and for your time @ChristianMurphy :)

I'll check with the Gatsby team

**Reply by wooorm** [reply]
btw: if gatsby-mdx supports `rehypePlugins` too (MDX does), you might try and see if `rehype-katex` works? Just an idea, not sure if it works

**Reply by nikoladev** [reply]
@wooorm [It does](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-plugin-mdx#rehype-plugins) and I did try it before, even toggling plugins one by one to see if anything sticks. But unfortunately it didn't. Thanks for thinking along :)

Though, seeing you here reminds me that I'm using `remark-math@3` because `gatsby-plugin-mdx` depends on `remark@10`. Off the top of your head, do you think that this could have anything to do with that combination? Would `remark@13` combined with `remark-math@4` magically solve this?

I've also [opened an issue at Gatsby's repo](https://github.com/gatsbyjs/gatsby/issues/28031).

**Reply by wooorm** [reply]
remark@13 w/ remark-math@4 would *not* solve this.
remark@10 works with remark-math@3. I believe everything, as Christian pointed out, is working okay on our side, but something in gatsby is acting up?

**Reply by nikoladev** [reply]
Yeah, I figured that was the case. Was just hoping for some deus ex machina :p

Thanks!

**Reply by wooorm** [reply]
I did see that several unified projects that Gatsby is using are from 3 years ago. E.g., `mdast-util-to-hast` is at v3, whereas we’re at v10 already

**Reply by nikoladev** [reply]
Thanks for investigating! I've mentioned it in [the issue](https://github.com/gatsbyjs/gatsby/issues/28031#issuecomment-726826198)
